### PR TITLE
Rephrase template documentation to be more clear about .gitea/template

### LINKED ISSUE #27336: Repository templates ignoring .gitea/template and including everything
### Description

Given a repository template ([demonstration template](https://try.gitea.io/andrewimeson/template)) with a `.gitea/template` file populated [according to the documentation](https://docs.gitea.com/usage/template-repositories), Gitea is ignoring the `.gitea/template` file and just including all files when you create from template ([demonstration template deploy repo](https://try.gitea.io/andrewimeson/from_template)). In the linked example files other than `README.md` and `.pre-commit-config.yaml` are included when deploying the template, which they shouldn't be.

### Gitea Version

1.20.4

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

Linux / Kubernetes

### How are you running Gitea?

Official Helm Chart v9.4.0

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by andrewimeson**
Closes #27336

### CODE REVIEW

**Review by nanguanlin6 — APPROVED** [review]

**Comment by lng2020** [comment]
Well, that's really confusing. When I reviewed this PR, I still misunderstood the meaning of `.gitea/template` and kept wondering why the files don't match glob rules are still included.

File: docs/content/usage/template-repositories.en-us.md [hunk 1]
```diff
@@ -17,9 +17,14 @@ menu:
 
 # Template Repositories
 
-Gitea `1.11.0` and above includes template repositories, and one feature implemented with them is auto-expansion of specific variables within your template files.
+Gitea (starting with version `1.11.0`) supports creating template repositories
+which can be used to generate repositories based on the template, complete with
+variable expansion of certain pre-defined variables.
 
-To tell Gitea which files to expand, you must include a `template` file inside the `.gitea` directory of the template repository.
+All files in the repository are included in a generated repository from the
```
Comment (andrewimeson) [inline docs/content/usage/template-repositories.en-us.md]: maybe it should be 
```suggestion
All files in the template repository are included in a generated repository from the
```

**Comment by andrewimeson** [comment]
@lng2020 I can rephrase it if you have suggestions. I am struggling to figure out how to phrase it, and it was already hard to understand.

**Comment by lng2020** [comment]
> @lng2020 I can rephrase it if you have suggestions. I am struggling to figure out how to phrase it, and it was already hard to understand.

I mean the name `.gitea/template` itself and its usage are confusing. Your PR looks good to me.

**Comment by lng2020** [comment]
Maybe we should explain "variable expansion" first? Otherwise it's easily ignored.

**Review by silverwind — APPROVED** [review]

**Pull request merged**